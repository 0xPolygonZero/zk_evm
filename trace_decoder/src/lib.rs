#![feature(linked_list_cursors)]
#![feature(trait_alias)]
#![feature(iter_array_chunks)]
// TODO: address these lints
#![allow(unused)]
#![allow(private_interfaces)]

mod compact;
pub mod decoding;
mod deserializers;
pub mod processed_block_trace;
pub mod trace_protocol;
pub mod types;
pub mod utils;
