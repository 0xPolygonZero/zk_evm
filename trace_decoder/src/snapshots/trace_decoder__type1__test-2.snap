---
source: trace_decoder/src/lib.rs
expression: executed
---
NonEmpty {
    inner: [
        Extension(
            Extension {
                key: NonEmpty {
                    inner: [
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        0,
                        1,
                        2,
                    ],
                },
                child: Branch(
                    Branch {
                        children: [
                            None,
                            None,
                            None,
                            Some(
                                Leaf(
                                    Leaf {
                                        key: NonEmpty {
                                            inner: [
                                                5,
                                            ],
                                        },
                                        value: Left(
                                            Value {
                                                raw_value: NonEmpty {
                                                    inner: [
                                                        49,
                                                        50,
                                                        51,
                                                        53,
                                                    ],
                                                },
                                            },
                                        ),
                                    },
                                ),
                            ),
                            Some(
                                Branch(
                                    Branch {
                                        children: [
                                            None,
                                            None,
                                            None,
                                            None,
                                            Some(
                                                Leaf(
                                                    Leaf {
                                                        key: NonEmpty {
                                                            inner: [
                                                                0,
                                                            ],
                                                        },
                                                        value: Left(
                                                            Value {
                                                                raw_value: NonEmpty {
                                                                    inner: [
                                                                        49,
                                                                        50,
                                                                        51,
                                                                        52,
                                                                    ],
                                                                },
                                                            },
                                                        ),
                                                    },
                                                ),
                                            ),
                                            Some(
                                                Leaf(
                                                    Leaf {
                                                        key: NonEmpty {
                                                            inner: [
                                                                0,
                                                            ],
                                                        },
                                                        value: Left(
                                                            Value {
                                                                raw_value: NonEmpty {
                                                                    inner: [
                                                                        49,
                                                                        50,
                                                                        51,
                                                                        52,
                                                                    ],
                                                                },
                                                            },
                                                        ),
                                                    },
                                                ),
                                            ),
                                            None,
                                            None,
                                            None,
                                            None,
                                            None,
                                            None,
                                            None,
                                            None,
                                            None,
                                            None,
                                        ],
                                    },
                                ),
                            ),
                            None,
                            None,
                            None,
                            None,
                            None,
                            None,
                            None,
                            None,
                            None,
                            None,
                            None,
                        ],
                    },
                ),
            },
        ),
    ],
}
